docs/src/peculiarities.md

```diff
@@ -0,0 +1,15 @@
+# Peculiarities
+
+## `_True` and `_False` (or `BoolsAsTypes`)
+
+When dispatch can be controlled by the value of a boolean variable, this introduces type instability.  Instead of introducing type instability, we chose to encode our boolean decision variables, like `threaded` and `calc_extent` in `apply`, as types.  This allows the compiler to reason about what will happen, and call the correct compiled method, in a stable way without worrying about 
+
+## What does `apply` return and why?
+
+@rafaqz
```

Comment:
```suggestion
`apply` returns the target geometries returned by `f`, whatever type/package they are from. It is recommended for consistency that `f` returns GeoInterface geometries unless there is a performance/conversion overhead to doing that. Objects that wrapped the target are replaced with GeoInterace.jl wrappers with matching `GeoInterface.trait` to the originals. All non-geointerface iterables become `Array`s.
```

Reply:
Hmm is that actually true about Tables? probably it should be, but DataFrame should return another DataFrame...

Reply:
yes, if the Table type provides a materializer, then we can convert - otherwise it returns a named tuple.